rcl_lifecycle/include/rcl_lifecycle/rcl_lifecycle.h

```diff
@@ -39,23 +88,70 @@ rcl_lifecycle_state_init(
   const char * label,
   const rcl_allocator_t * allocator);
 
+/// Finalize a rcl_lifecycle_state_t.
+/**
+ *
+ * Calling this will set the rcl_lifecycle_state_t struct into the an unitialized state that is
+ * functionally the same as before rcl_lifecycle_state_init was called.
```

Comment:
This is truth but I think the wording should be changed to explicitly note that this method will "close" the structure by freeing the resources taken by the init allocator.

Reply:
Something similar to this [rcl timer example](https://github.com/ros2/rcl/blob/8a0ec5fe959680c18e599fb4580813ddf14fe352/rcl/include/rcl/timer.h#L163)